src/librustc_mir/build/matches/simplify.rs

```diff
@@ -196,8 +195,11 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                 Ok(())
             }
 
-            PatKind::Or { .. } => {
-                Err(match_pair)
+            PatKind::Or { ref pats } => {
+                candidate.match_pairs.extend(
+                    pats.iter().map(|pat| MatchPair::new(match_pair.place.clone(), pat)),
```

Comment:
We should maybe leave a FIXME here to remove this once we use a strategy that doesn't result in exponential blow-up? Although this is perhaps the strategy to use for `let` statements.

Reply:
It's linear in the number of patterns, so I don't think it ought to create blowup.

Reply:
Hmm, but what happens if you have:
```rust
match (x, y, z) {
    (0 | 1 | 2, 0 | 1 | 2, 0 | 1 | 2) => {}
    _ => {}
}
```
Won't this simplify to 9 tests in sequence (as if you wrote down the "DNF" of the above) as opposed to testing `x in 0..=2` first, then `y in 0..=2`, and then `z in 0..=2`?

(https://github.com/rust-lang/rfcs/blob/master/text/2535-or-patterns.md#implementation-notes)